execute/plugin.go

```diff
@@ -400,8 +405,13 @@ func (p *Plugin) Outcome(
 				if msg, ok := observation.Messages[report.SourceChain][i]; ok {
 					report.Messages = append(report.Messages, msg)
 				}
+
```

Comment:
Is the `i` correct here? (I think so)

```
for i, report := range commitReports {
			report.Messages = nil
			for i := report.SequenceNumberRange.Start(); i <= report.SequenceNumberRange.End(); i++ {
```

I believe second loop should use `j` to prevent using the `i` from the outer loop.